cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -265,6 +265,9 @@ if(WIN32)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /LARGEADDRESSAWARE")
 
     if (CMAKE_COMPILE_WARNING_AS_ERROR)
+        if (CMAKE_VERSION VERSION_LESS 3.24)
+            ie_add_compiler_flags(/WX)
```

Comment:
can we also enabled linker warnings as errors ? See https://learn.microsoft.com/en-us/cpp/build/reference/wx-treat-linker-warnings-as-errors?view=msvc-170

Because we still have warnings from linker like:
```
[1797/7103] Linking CXX static library C:\a\1\openvino\bin\intel64\Release\ie_docs_snippets.lib
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
ov_properties_migration.cpp.obj : warning LNK4006: "int __cdecl main_new(void)" (?main_new@@YAHXZ) already defined in ov_properties_migration.cpp.obj; second definition ignored
ov_properties_migration.cpp.obj : warning LNK4006: "int __cdecl main_old(void)" (?main_old@@YAHXZ) already defined in ov_properties_migration.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
custom_op.cpp.obj : warning LNK4006: main already defined in custom_op.cpp.obj; second definition ignored
```